lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -612,13 +611,20 @@ defmodule ExUnit.Assertions do
     |> elem(1)
   end
 
+  defp expand_pattern({:when, meta, [left, right]}, caller) do
+    left = expand_pattern_except_vars(left, Macro.Env.to_match(caller))
+    right = expand_pattern_except_vars(right, %{caller | context: :guard})
```

Comment:
Should we have a corresponding `Macro.Env.to_guard/1`?

Reply:
We created to_match because we were doing in that in 3 or 4 different places. I think to_guard is a good idea... but right now this will be its second use. And one of them is inside Kernel.Utils, where we can't invoke Macro.Env for bootstrap reasons. :(